### ISSUE TITLE
Unable to parse statement Caused by: java.io.IOException: Mark invalid

### ISSUE THREAD

**Issue by hauntingEcho** [flyway/flyway#2928]
###### Which version and edition of Flyway are you using?

I've confirmed this issue using flyway 6.5.5 on Windows.  I am not able to reproduce the issue on 6.5.4 or 6.5.5 on Linux, and do not have access to OSX for testing.  Teammates have reported the issue on 6.5.5 & 6.5.4 on OSX, and 6.5.5, 6.5.4, and 6.5.1 on Windows.  All of these are the Community Edition CLI.

This issue can be worked around by downgrading to 6.4.4 on any OS

###### Which database are you using (type & version)?

Postgres 12.3 on AWS RDS

###### What did you do?

assuming you've extracted the attached zip file in your current folder:
[migrations.zip](https://github.com/flyway/flyway/files/5211565/migrations.zip)

`flyway migrate migrate -locations="filesystem:migrations" -schemas="${SCHEMA}" -url="${URL}"`

Note that I haven't been able to reproduce the issue when making ANY change to the file (including deleting blank lines or comments!)

###### What did you expect to see?

success

###### What did you see instead?

```
ERROR: Unable to parse statement in migrations\V1__failure.sql at line 25 col 1. See https://flywaydb.org/documentation/knownparserlimitations for more information: Mark invalid
Caused by: java.io.IOException: Mark invalid
```

**Comment by lbreuss** [comment]
I can confirm, that this problem just recently also surfaced in my Ubuntu 18.04 VM. 

I was using flyway 6.5.5 for many month now, and the migration file written 2 years ago has never been changed. 
The "offending" DROP SEQUENCE statement is followed by an empty line and 3 comment lines, before the next ALTER TABLE statement starts. A very similar block 10 lines above is parsed without error...

Just this week it started throwing "Mark invalid" in that particular migration. Possibly after a regular OS and Software update. Java has been updated 1 month prior to this problem starting, should not be the immediate problem. (OpenJDK 64-Bit Server VM (build 11.0.10+9-Ubuntu-0ubuntu1.18.04, mixed mode, sharing)

Hope it's fixed in 6.5.7

**Comment by hauntingEcho** [comment]
Flyway 6.5.6 fixed this for me if I recall correctly (whichever version was after @DoodleBobBuffPants 's work linked above)
